build.sh

```diff
@@ -98,6 +92,17 @@ else
     export STEAM_PASSWORD=${STEAM_PASSWORD:-}
 fi
 
+# Process user variables
+REGISTRY_GOLDSOURCE=${REGISTRY_GOLDSOURCE:?err}
+REGISTRY_SOURCE=${REGISTRY_SOURCE:?err}
+if [ ! "$NO_PUSH" = 'true' ]; then
+    REGISTRY_USER=${REGISTRY_USER:?err}
+    REGISTRY_PASSWORD=${REGISTRY_PASSWORD:?err}
+else
+    REGISTRY_USER=${REGISTRY_USER:-}
+    REGISTRY_PASSWORD=${REGISTRY_PASSWORD:-}
+fi
```

Comment:
no reason for this section to be moved below

Reply:
it needs NO_PUSH variable to exist

Reply:
sort them in the original order